src/hal/common.c

```diff
@@ -80,14 +80,15 @@ int universal_i2c_write_register(int fd, unsigned char i2c_addr,
                                  unsigned int reg_addr, unsigned int reg_width,
                                  unsigned int data, unsigned int data_width) {
     (void)i2c_addr;
-    (void)data;
-    char buf[2];
+    char buf[3];
 
     if (reg_width == 2) {
         buf[0] = (reg_addr >> 8) & 0xff;
         buf[1] = reg_addr & 0xff;
+        buf[2] = data;
```

Comment:
Can you please describe the purpose of the changes? I don't remember exactly, but it seems, previous version worked on all platforms without it

Reply:
It sets the data that needs to be written to the i2c device.
I can revert the changes in this case since it only needs to write 0x0, which should be the default value of buf[1].

Reply:
It's clear now